Controller/Adminhtml/Configuration/MerchantAccounts.php

```diff
@@ -0,0 +1,90 @@
+<?php
+
+namespace Adyen\Payment\Controller\Adminhtml\Configuration;
+
+use Adyen\AdyenException;
+use Adyen\Payment\Helper\ManagementHelper;
+use Magento\Framework\App\Action\Action;
+use Magento\Framework\Controller\Result\JsonFactory;
+use Magento\Framework\App\Action\Context;
+
+class MerchantAccounts extends Action
+{
+
+    const TEST_MODE = 'test';
+    const LIVE_MODE = 'production';
+    /**
+     * @var ManagementHelper
+     */
+    protected $managementHelper;
+
+    /**
+     * @var JsonFactory
+     */
+    protected $resultJsonFactory;
+
+    /** @var \Magento\Framework\App\Request\Http */
+    protected $request;
+
+    /**
+     * @var \Adyen\Payment\Helper\Data
+     */
+    protected $_adyenHelper;
+
+    public function __construct(
+        Context $context,
+        ManagementHelper $managementHelper,
+        JsonFactory $resultJsonFactory,
+        \Magento\Framework\App\Request\Http $request,
+        \Adyen\Payment\Helper\Data $adyenHelper
+    ) {
+        parent::__construct($context);
+        $this->resultJsonFactory = $resultJsonFactory;
+        $this->managementHelper = $managementHelper;
+        $this->request = $request;
+        $this->_adyenHelper = $adyenHelper;
+    }
+
+    /**
+     * @return \Magento\Framework\Controller\Result\Json
+     */
+    public function execute()
+    {
+        $resultJson = $this->resultJsonFactory->create();
+        try {
+            $xapikey = $this->getRequest()->getParam('xapikey', '');
+            $demoMode = $this->getRequest()->getParam('demoMode', '');
+            if (str_contains($xapikey, '*')) {
```

Comment:
This is checking if the value from the frontend is masked/obscured no? If so I think this would ignore real API keys that happen to have a * character.